prelude/bootstrap.js

```diff
@@ -565,10 +569,53 @@ function payloadFileSync(pointer) {
   // open //////////////////////////////////////////////////////////
   // ///////////////////////////////////////////////////////////////
 
+  function removeTemporaryFolderAndContent(folder) {
+    if (NODE_VERSION_MAJOR <= 14) {
+      if (NODE_VERSION_MAJOR <= 10) {
+        // folder must be empty
+        for (const f of fs.readdirSync(folder)) {
+          fs.unlinkSync(path.join(folder, f));
+        }
+        fs.rmdirSync(folder);
+      } else {
+        fs.rmdirSync(folder, { recursive: true });
+      }
+    } else {
+      fs.rmSync(folder, { recursive: true });
+    }
+  }
+  const temporaryFiles = {};
+  const os = require('os');
+  const tmpFolder = fs.mkdtempSync(path.join(os.tmpdir(), 'pkg-'));
+  process.on('beforeExit', () => {
+    removeTemporaryFolderAndContent(tmpFolder);
```

Comment:
Is this really needed? Couldn't we keep files in temporary folder for quicker startup?

Reply:
-This is a boy scout attitude. Leave the place tidier than you found it.
- Those files potentially are exposed in clear, 
- Arguably, those files could be corrupted later on somehow, the single source of truth is in the packed binary.
- Only files that are  randomly accessed are copied uncompressed in this folder. I made sure that file that are read globally (readFileSync) or streamed (createReadStream) are processed in memory.  I would be inclined to leave it this way unless users complain about potential performance issue.
- having a persistent cache would require that the temp subfolder name is stable and unique, based on the executable signature (shall we calculated the md5/sha1 of the entire pkg.exe ?)

Reply:
> having a persistent cache would require that the temp subfolder name is stable and unique, based on the executable signature (shall we calculated the md5/sha1 of the entire pkg.exe ?)

Could be an idea but yeah those files could be someway corrupted. I would like to also know @jesec and @leerob opinion about this

Reply:
It might also be an option to generate a build-id uuid on build and bake that into the binary.